oak/common/utils.cc

```diff
@@ -43,5 +44,46 @@ void write_file(const std::string& data, const std::string& filename) {
   t.close();
 }
 
+std::map<std::string, std::string> read_pem(const std::string& filename) {
```

Comment:
I haven't really read through, but does https://github.com/google/boringssl/blob/master/include/openssl/pem.h have something useful perhaps? Or some other standard library for parsing PEM files? If not, could you add some tests to this logic?